expression/builtin_arithmetic.go

```diff
@@ -416,37 +416,48 @@ func (s *builtinArithmeticMinusIntSig) evalInt(row chunk.Row) (val int64, isNull
 		return 0, isNull, err
 	}
 	forceToSigned := s.ctx.GetSessionVars().SQLMode.HasNoUnsignedSubtractionMode()
-	isLHSUnsigned := !forceToSigned && mysql.HasUnsignedFlag(s.args[0].GetType().Flag)
-	isRHSUnsigned := !forceToSigned && mysql.HasUnsignedFlag(s.args[1].GetType().Flag)
+	isLHSUnsigned := mysql.HasUnsignedFlag(s.args[0].GetType().Flag)
+	isRHSUnsigned := mysql.HasUnsignedFlag(s.args[1].GetType().Flag)
 
-	if forceToSigned && mysql.HasUnsignedFlag(s.args[0].GetType().Flag) {
-		if a < 0 {
-			return 0, true, types.ErrOverflow.GenWithStackByArgs("BIGINT UNSIGNED", fmt.Sprintf("(%s - %s)", s.args[0].String(), s.args[1].String()))
+	switch {
+	case forceToSigned && isLHSUnsigned && isRHSUnsigned:
+		ua, ub, uMaxInt64, uMinInt64 := uint64(a), uint64(b), uint64(math.MaxInt64), uint64(math.MaxInt64+1)
+		if (ua >= ub && ua-ub > uMaxInt64) || (ua < ub && ub-ua > uMinInt64) {
+			return 0, true, types.ErrOverflow.GenWithStackByArgs("BIGINT", fmt.Sprintf("(%s + %s)", s.args[0].String(), s.args[1].String()))
```

Comment:
Why this error message is "(%s + %s)"?

Reply:
I reused the report format from [vec version](https://github.com/pingcap/tidb/blob/master/expression/builtin_arithmetic_vec.go#L388).

```
if lh < 0 {
	return types.ErrOverflow.GenWithStackByArgs("BIGINT UNSIGNED", fmt.Sprintf("(%s - %s)", b.args[0].String(), b.args[1].String()))
}

if rh < 0 {
	return types.ErrOverflow.GenWithStackByArgs("BIGINT UNSIGNED", fmt.Sprintf("(%s - %s)", b.args[0].String(), b.args[1].String()))
}

if (lh > 0 && -rh > math.MaxInt64-lh) || (lh < 0 && -rh < math.MinInt64-lh) {
	return types.ErrOverflow.GenWithStackByArgs("BIGINT", fmt.Sprintf("(%s + %s)", b.args[0].String(), b.args[1].String()))
}
```



After recheck I think we found another bug, `minusFUU`, `minusFUS`, `minusFSU` all report in `%s+%s` format.

Reply:
Changed `%s + %s` to `%s - %s` in row version eval.